Added tutorial doc on how to add a Space to Arxiv

abidlabs opened the pull request:
Adds a step-by-steps post on how to add a Space to ArXiv. Closes: #526

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/hub/pr_530). All of your documentation changes will be reflected on that endpoint.

osanseviero commented on docs/hub/arxiv.md:
````diff
@@ -0,0 +1,40 @@
+    layoutlm_dummy = LayoutLMForTokenClassification.from_pretrained("microsoft/layoutlm-base-uncased", num_labels=1)
+    ```
+
+    More instructions on [building a demo on Hugging Face Spaces are here](./spaces-overview). 
````
Maybe you can add something that says that you can use `huggingface_hub` to download a model programmatically. For most cases they won't have out of the box integration with a library

julien-c replied:
you can also link to the page where we describe that if automatic model/dataset detection does not work you can specify them explicitly in the README metadata

osanseviero commented on docs/hub/arxiv.md:
```diff
@@ -0,0 +1,40 @@
+
+4. That's it! As soon as your Space is built, Hugging Face will detect that it is associated with the model and your Space should appear in the Demo tab on ArXiv in a few minutes.
+
+    *Note*: after your Space is built, you can verify that it is linked to your model. A "Linked Models" button should appear in the top right corner of the Space and link to the model. 
```
Add a screenshot here

osanseviero reviewed (commented):
Very cool! I left some comments

julien-c commented on docs/hub/_toctree.yml:
```diff
@@ -135,7 +135,9 @@
     - local: spaces-github-actions
-      title: Managing Spaces with Github Actions      
+      title: Managing Spaces with Github Actions 
+    - local: arxiv
```
```suggestion
    - local: spaces-add-to-arxiv
```

julien-c commented on docs/hub/arxiv.md:
```diff
@@ -0,0 +1,40 @@
+learning research without writing any code. Hugging Face and ArXiv have collaborated 
+to embed these demos directly along side papers on ArXiv!
+
+Thanks to this integration, users can now find the most popular demos for a paper on its arXiv abstract page. For example, if you want to try out demos of the LayoutLM document classification model, you can go to [LayoutLM's arXiv page](https://arxiv.org/abs/1912.13318), and navigate to the demo tab. You will see open-source demos built by the machine learning community for this model.
```
```suggestion
Thanks to this integration, users can now find the most popular demos for a paper on its arXiv abstract page. For example, if you want to try out demos of the LayoutLM document classification model, you can go to [the LayoutLM paper's arXiv page](https://arxiv.org/abs/1912.13318), and navigate to the demo tab. You will see open-source demos built by the machine learning community for this model.
```

julien-c commented on docs/hub/arxiv.md:
```diff
@@ -0,0 +1,40 @@
+**Prerequisites**
+
+* There's an existing paper on ArXiv that you'd like to create a demo for
+* The paper is **associated with a model** that is on the Hugging Face Model Hub (or can be uploaded there)
```
```suggestion
* The paper is **associated with a model or a dataset** that is on the Hugging Face Model Hub (or can be uploaded there)
```

julien-c commented on docs/hub/arxiv.md:
```diff
@@ -0,0 +1,40 @@
+
+* There's an existing paper on ArXiv that you'd like to create a demo for
+* The paper is **associated with a model** that is on the Hugging Face Model Hub (or can be uploaded there)
+* You have built or (can build) a demo for the model on Spaces
```
```suggestion
* You have built or (can build) a demo for the model or the dataset on Spaces
```

julien-c commented on docs/hub/arxiv.md:
```diff
@@ -0,0 +1,40 @@
+
+**Steps to Add Your Space**
+
+1. First, upload the model associated with the ArXiv paper onto the Hugging Face Model Hub if it is not already there. ([Detailed instructions are here](./models-uploading))
```
```suggestion
1. First, upload the model or dataset associated with the ArXiv paper onto the Hugging Face Hub if it is not already there. ([Detailed instructions are here](./models-uploading))
```

julien-c commented on docs/hub/arxiv.md:
```diff
@@ -0,0 +1,40 @@
+
+    *Note*: you can verify this step has been carried out successfully by seeing if an ArXiv button appears above the model card. In the case of LayoutLM, the button says: "arxiv:1912.13318" and links to the LayoutLM paper on ArXiv.
+
+3. Then, create a demo on Spaces that loads this model. Somewhere within the code, the full path to the model must be included in order for Hugging Face to detect that a Space is associated with a particular model.
```
```suggestion
3. Then, create a demo on Spaces that loads this model or dataset. Somewhere within the code, the model or dataset Hub id must be included in order for Hugging Face to detect that a Space is associated with it.
```

julien-c reviewed (commented):
My two feedbacks:
- inter-link more to other doc pages, e.g. the one on Space README.md metadata
- this is not just for models but also for datasets. As we try to push for more recognition for datasets (including in published papers) would be cool to highlight both

abidlabs commented:
Thanks for the great suggestions @osanseviero and @julien-c! 

* Updated the URL to `spaces-add-to-arxiv`
* Added references to datasets
* Added more inter-linking
* Added screenshots

osanseviero reviewed (approved):
Looks nice in https://moon-ci-docs.huggingface.co/docs/hub/pr_530/en/spaces-add-to-arxiv :fire:

osanseviero commented:
Feel free to merge if you're happy with it :)

abidlabs commented:
Thanks @osanseviero!

julien-c reviewed (commented):
yay! quite cool

julien-c commented:
maybe we could have added a screenshot of what it looks like on Arxiv and/or a link to the blogpost where we announced the integration

abidlabs commented:
Sure thanks for the suggestion @julien-c, will add later this week
